utils/state.py

```diff
@@ -61,3 +69,17 @@ def add(self, key):
                 f"[state] key {key} already exists in {self.state_path}")
         self.client.put_object(
             Bucket=self.bucket, Key=f"{self.state_path}/{key}")
+
+    def rm(self, key):
```

Comment:
`__delitem__`? so on can do `del(state[key])`

Reply:
i chose not to use `__setitem__` because (currently) the state only stores keys, without values. that's why using `state.add()` makes more sense to me, rather then `state[key] = 'whatever'`.

for this reason for me it feels less intuitive to use `state[key]` at this point. the dictionary feeling does not represent reality as of yet.

Reply:
Ok, got it.